main.go

```diff
@@ -86,27 +86,30 @@ func main() {
 	}
 
 	// TODO: configure session
+	awsRegion := os.Getenv("AWS_REGION")
```

Comment:
Can we infer the region from the EKS cluster if the environment variable is not set? This would allow customers to avoid having to modify files in order to install the controller.

Reply:
Also, AWS [go SDK](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-the-region) already gets the region from AWS_REGION environment variable. Perhaps we can infer the region from the cluster as a fallback if the SDK itself cannot determine region.

Reply:
>This would allow customers to avoid having to modify files in order to install the controller

The customer won't have to modify any files, they need to just to set this variable `export AWS_REGION=us-west-2` before applying `kubectl apply -k "github.com/aws/aws-cloud-map-mcs-controller-for-k8s/config/controller_install_release"` or even the following command will work too `AWS_REGION=us-west-2 kubectl apply -k "github.com/aws/aws-cloud-map-mcs-controller-for-k8s/config/controller_install_release"`

Reply:
> Can we infer the region from the EKS cluster if the environment variable is not set?

We also need to support non-eks cluster setup, and cross region setup.

> AWS go SDK already gets the region from AWS_REGION environment variable

We would still need make this environment variable "available" inside the controller container. Let me tackle fallback separately.

Reply:
Ok. What happens if `kubectl apply -k ...` is run without setting the environment variable? Does the controller still start? What does the value of the environment variable become?

Reply:
Yes, the controller will start. But we will see below error in the controller logs. I haven't deployed in eks cluster though.
```
"operation error ServiceDiscovery: ListNamespaces, failed to resolve service endpoint, an AWS region is required, but was not found"
```
We are talking about implementing fail-fast in case region or credentials are missing.

Reply:
Thanks for checking.